pyproject.toml

```diff
@@ -0,0 +1,2 @@
+[build-system]
+requires = ["pip>=18.1", "setuptools>=40.8", "wheel", "numpy"]
```

Comment:
@oesteban This should allow us to install purely with `pip install .`, as `pip` will parse this first to make sure it has the build dependencies.

pip 18.1 allows us to use a dependency that looks like `"package @ URL"`, so we can handle all dependencies directly in `install_requires`.

Reply:
Relevant discussion: https://stackoverflow.com/a/53412651

Possibly the reason this is failing for me is that pip removed dependency_links in version 19.

Reply:
Hmm. Looks like you can't actually set a minimum version on pip in this way. https://github.com/pypa/pip/issues/4145

So this would mean we need to prepare for users with old pip, which is likely on most HPCs.